src/NServiceBus.Core/Transports/TransportReceiver.cs

```diff
@@ -61,19 +63,29 @@ class TransportReceiver
 
         async Task InvokePipeline(PushContext pushContext)
         {
+            var pipelineStartedAt = DateTime.UtcNow;
+
             using (var childBuilder = builder.CreateChildBuilder())
             {
                 var rootContext = new RootContext(childBuilder, pipelineCache, eventAggregator);
-                var context = new TransportReceiveContext(new IncomingMessage(pushContext.MessageId, pushContext.Headers, pushContext.BodyStream), pushContext.TransportTransaction, pushContext.ReceiveCancellationTokenSource, rootContext);
+                
+                var message = new IncomingMessage(pushContext.MessageId, pushContext.Headers, pushContext.BodyStream);
+                var context = new TransportReceiveContext(message, pushContext.TransportTransaction, pushContext.ReceiveCancellationTokenSource, rootContext);
+
                 context.Extensions.Merge(pushContext.Context);
+
                 await pipeline.Invoke(context).ConfigureAwait(false);
 
-                await context.RaiseNotification(new ReceivePipelineCompleted()).ConfigureAwait(false);
+                //notifications are only relevant for the main pipeline since satellites pipelines are not exposed to user in any way
+                if (isMainReceiver)
+                {
+                    await context.RaiseNotification(new ReceivePipelineCompleted(message, pipelineStartedAt, DateTime.UtcNow)).ConfigureAwait(false);
+                }
             }
         }
 
+        bool isMainReceiver;
```

Comment:
Really nitpicker, since we have already a bool flag in this class how about moving it down to the other bool flag?